build-and-test.ps1

```diff
@@ -50,20 +54,16 @@ function Get-ImageBuilder()
 
 $imageBuilderFolder = [System.IO.Path]::Combine("$PSScriptRoot", ".Microsoft.DotNet.ImageBuilder")
 $imageBuilderExe = [System.IO.Path]::Combine("$imageBuilderFolder", "image-builder", "Microsoft.DotNet.ImageBuilder.exe")
-if (-not (Test-Path -Path "$imageBuilderExe" -PathType Leaf))
-{
+if (-not (Test-Path -Path "$imageBuilderExe" -PathType Leaf)) {
     Get-ImageBuilder
 }
 
-$imageBuilderArgs = "build --path $buildFilter "
-if ($IsDryRun)
-{
+$imageBuilderArgs = "build --path $buildFilter --test-var VersionFilter=$VersionFilter --test-var OSFilter=$OSFilter"
+if ($IsDryRun) {
```

Comment:
What do you think about replacing the IsDryRun parameter with a string CustomImageBuilderArgs?  This would make this script more robust so that it could be used within the official builds.  The CustomImageBuilderArgs could be used to specify the `push` related arguments.

Reply:
To clarify - change the type of the parameter to `string`, and append the parameter value to the default args.  So in that case, I think `ImageBuilderCustomArgs` would convey better, and the logic would be -

```
$imageBuilderArgs = "build --path $buildFilter --test-var VersionFilter=$VersionFilter --test-var OSFilter=$OSFilter"
if ([string]::IsNullOrWhiteSpace($ImageBuilderCustomArgs)) {
    $imageBuilderArgs += " $ImageBuilderCustomArgs"
}
```

Reply:
yes that is what I was thinking - and I like your suggested name.